### ISSUE TITLE
Notebook creation UI missing validation on "custom image" field

### ISSUE THREAD

**Issue by djwhatle** [kubeflow/kubeflow#6621]
/kind bug

**What steps did you take and what happened:**

1. Filled out the `custom image` field of the Kubeflow Notebook creation UI with an invalid image URI with a leading space.
2. Created a notebook
3. After creation, notebook UI shows loading spinner forever, events revealed StatefulSet creation was failing in notebook controller

```
$ kubectl get event --sort-by='.lastTimestamp' -n my-ns

LAST SEEN   TYPE      REASON                         OBJECT                                          
13m         Warning   FailedCreate                   statefulset/a100x16                             

MESSAGE
create Pod a100x16-0 in StatefulSet a100x16 failed error: Pod "a100x16-0" is invalid: 
spec.containers[0].image: Invalid value: " gcr.io/leading-space-example": 
must not have leading or trailing whitespace
```

![image](https://user-images.githubusercontent.com/7576968/185943709-622fe0c7-1439-4a59-aae8-b52ee8bdcd90.png)

![image](https://user-images.githubusercontent.com/7576968/185945949-67623d1d-9ae1-4ea7-9615-8f7c8121a901.png)

**What did you expect to happen:**

Custom image field is validated according to a regex

**Environment:**

- Kubeflow version:  1.4

**Comment by juliusvonkohout** [comment]
Yes, the controller is not resilient. Do you want to help fixing it?

**Comment by andreyvelich** [comment]
This should be fixed in #6529
